### PR TITLE
Make the DOI Resolution Fetcher return nothing when the DOI leads to a host for which a tailored fetcher exists

### LINKED ISSUE #6922: DoiResolution Fetcher fetches whole book for some Springer conference papers
JabRef version 5.1 on OpenSUSE Leap 15.2

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Have the conference paper [Map2Check Using LLVM and KLEE](https://doi.org/10.1007/978-3-319-89963-3_28) in your library
2. "Search full text documents online"
3. JabRef downloads a PDF of the whole conference proceedings instead of just the paper

I looked through the code a bit and it seems like the `DoiResolution` fetcher has a higher trust value (`TrustLevel.SOURCE`) than the `SpringerLink` fetcher (`TrustLevel.PUBLISHER`). Whereas the `SpringerLink` fetcher gets the correct link to the paper PDF, the `DoiResolution` fetcher selects the wrong PDF link from the Springer website.

### PR DESCRIPTION
**Pull request by Toromtomtom**
Fixes #6922

- [ ] Change in CHANGELOG.md described (not applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by tobiasdiez** [comment]
As you noticed, the underlying problem is actually that the DOI fetcher has a higher trust value as the publishers. I think it would be a good idea to change it to "publishers > identifier-based resolution (doi, arXiv) > general search (google)". @JabRef/developers @Toromtomtom  do you see any problem with this solution?

**Comment by Toromtomtom** [comment]
> I think it would be a good idea to change it to "publishers > identifier-based resolution (doi, arXiv) > general search (google)".

I also think that this would be a better solution.

**Comment by koppor** [comment]
+1 from my side, too

**Comment by Toromtomtom** [comment]
I reverted my previous commits and decreased the trust level of the DOI resolution fetcher. This works for me, but maybe someone more involved in the project wants to weigh in on the ranking of the full text fetchers.

**Review by Siedlerchr — APPROVED** [review]

**Comment by koppor** [comment]
Food for thought:

- A DOI uniquely identifies a paper. Per defition, a DOI leads to the right paper. Everything else is good guessing.
- One title of a paper may lead to different publications of it. One the confernce version, the other the journal version. --> the PDF could be chosen randomly
- What about the consequences for other fetchers? Do we overlook something?
- Can't we contact Springer to fix their DOI 2 PDF mapping?

Proposal: Can we add a special handling for Springer? If a DOI directs to Springer, we use the Springer Fetcher. In all other cases, the functionality is untouched. In this way, we accept that this is a hack.

To really judge, there would be a test needed retrieving 1000 papers and check whether the retrieval rate is higher or lower with this check. - Alternatively, can we add telemetry for that?

File: src/main/java/org/jabref/logic/importer/fetcher/DoiResolution.java [hunk 1]
```diff
@@ -121,6 +121,6 @@
 
     @Override
     public TrustLevel getTrustLevel() {
-        return TrustLevel.SOURCE;
+        return TrustLevel.META_SEARCH;
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fetcher/DoiResolution.java:124]: This decreases the result quality of the DOI fetcher (always leading to the "right" paper) to the quality of Google Scholar. (From the highest to the lowest)

Can the solution of the title? 😇 

> Make the DOI Resolution Fetcher return nothing when the DOI leads to a host for which a tailored fetcher exists

**Comment by Siedlerchr** [comment]
The problem is that a DOI often does not lead to the fulltext version directly, but to the site where to find the fulltext. And our DOIResolution Fetcher does some magic guessing by looking at the first PDF-link the sourcecode of the website. 
Every publisher's/journal page looks different.

**Comment by tobiasdiez** [comment]
The springer fetcher also only looks at the DOI, but uses the springer API to find the correct URL for the download.
https://github.com/JabRef/jabref/blob/deb2f20bac50e584d466aa00cdbfe9972d2c9c09/src/main/java/org/jabref/logic/importer/fetcher/SpringerLink.java#L39-L49

**Comment by koppor** [comment]
Devcall decision: Use first solution. -- @koppor will do git magic

**Comment by Toromtomtom** [comment]
All right, thanks for taking care of this!

**Comment by tobiasdiez** [comment]
@koppor In addition, the SpringerLink should have a higher trust score as the DoiResolution fetcher, since it's also DOI-based but custom-tailored to Springer. I would also merge this class with the other springer fetcher.

**Comment by Toromtomtom** [comment]
Is there anything I can do to move this forward? Reset the branch or something?

**Comment by koppor** [comment]
Steps:

1. I do as promised at https://github.com/JabRef/jabref/pull/6937#issuecomment-700482658
2. I search all issues and PRs to document the idea of the concept (priorities, information sources, ...)
3. I think how to transform the comment https://github.com/JabRef/jabref/pull/6937#issuecomment-700629524 to a follow-up issue. Which implications should be thought of.

In parallel, I discuss with @stefan-kolb, because he invented the whole thing. My mistake was not to enforce that design decisions are documented (either as ADR or as other text files)

**Comment by stefan-kolb** [comment]
Initial PR https://github.com/JabRef/jabref/pull/3882

**Comment by koppor** [comment]
I think, I collected all documentation and put it at the appropriate place at https://github.com/JabRef/jabref/pull/6990.

So, nothing to do for @Toromtomtom in this PR.

**Comment by koppor** [comment]
The first two commits are in master now. See https://github.com/JabRef/jabref/commit/ce9f714f6d01c8a49fccf5c8df9cbf5bc3e8403f.

**Pull request closed without merging**